[@types/parse] Major version bump to 2.0.0 due to Parse 2.1.0

agoldis opened the pull request:
__Checklist__
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

- https://parseplatform.org/Parse-SDK-JS/api/2.1.0/
- https://github.com/parse-community/Parse-SDK-JS/tree/abdbda17ee31c2b53157c73ef3306377fb940620

- [x] Increase the version number in the header if appropriate. 
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.


__Notes__
i have created a new major version of `@typed/parse`, following the guide:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#i-want-to-update-a-package-to-a-new-major-version

The old version (`v1`) defines obsolete entities, that are not implemented in https://www.npmjs.com/package/parse (version 2.1.0), it was also dependant on older type definitions, causing errors for users. (See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/29273)

The non-existing entities were safely removed together with the associated tests and dependencies, to be exact, here's what was removed:
- dependency on `jquery`
- dependency on `underscore`
- `Parse.History`
- `Parse.Collection`
- `Parse.Events`
- `Parse.Router`
- `Parse.View`

I also removed references to `Backbone`-related terms from in-code documentation
There are no new definitions or changes to the definitions that were not removed

Resolves https://github.com/DefinitelyTyped/DefinitelyTyped/issues/29273

__Mentions__
Ullisen Media Group <http://ullisenmedia.com>
@dpoetzsch David Poetzsch-Heffter <https://github.com/dpoetzsch>
Cedric Kemp <https://github.com/jaeggerr>
@flavionegrao Flavio Negrão <https://github.com/flavionegrao>
@wesleygrimes Wes Grimes <https://github.com/wesleygrimes>
Otherwise SAS <https://github.com/owsas>

armanio123 commented:
@dpoetzsch @jaeggerr @flavionegrao @wesleygrimes @owsas would any of you mind reviewing this PR?

agoldis commented:
@wesleygrimes thanks for volunteering to review the PR, please tell me if you need any assistance with  it - I will be glad to describe what I have changed.

wesleygrimes commented on types/parse/parse-tests.ts:
```diff
@@ -545,17 +470,3 @@ function test_batch_operations() {
     Parse.Object.fetchAll(games, { sessionToken: '' })
     Parse.Object.fetchAllIfNeeded(games, { sessionToken: '' })
 }
-
```
@agoldis any particular reason this test was removed. Parse.Query should still be a valid type in the new version correct?

agoldis replied:
are you talking about `test_query_subscribe`? 
> any particular reason this test was removed
The test checks `query.subscribe` functionality, which doesn't exist in the new version.
> Parse.Query should still be a valid type in the new version correct?
Yes, `Parse.Query` is still there, but some methods has gone, e.g. `subscribe`

wesleygrimes replied:
Can you point to me to the documentation that states query.subscribe was deprecated? 
Subscribing is a core feature of live queries in parse server which are basically a wrapper around a web socket implementation.

agoldis replied:
oh 🙀 you're right! good catch! I will fix that and will let you know.

wesleygrimes replied:
Ok great thanks! Can you make sure everything else is there that should be as well? I’m glad you’re getting us upgraded to the latest. Just want to get these few things ironed out and then we can probably get this merged in. Thanks!

agoldis commented:
@wesleygrimes 
I have updated the PR:
- added `Parse.Query.subscribe` as appears https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseQuery.js#L1389
- added `Parse.LiveQuerySubscription` as appears https://github.com/parse-community/Parse-SDK-JS/blob/master/src/LiveQuerySubscription.js#L89
- Re-enabled and adjusted `test_query_subscribe`

I verified again that the rest of removed entities are not implemented anymore - checked [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/Parse.js) and [here](https://parseplatform.org/Parse-SDK-JS/api/2.1.0/index.html)

wesleygrimes commented:
@agoldis it's looking better. I think the last thing we need to look at is the `Promise` type.  According to parse's documentation `Parse.Promise` has been removed and replaced with native `Promise`. Can you update the typings and tests to reflect this?

agoldis commented:
@wesleygrimes updated
@dpoetzsch Please note the changes in `parse-mockdb` to rewire dependencies and module structure

wesleygrimes commented:
@agoldis I think we are good, but I would feel more comfortable if another contributor (@dpoetzsch) could do a final review. Then we can get this merged. Thanks for all your hard work on this. Will be nice to get the typings updated.

armanio123 commented:
Seems like there's no review yet from @dpoetzsch , would you like me to merge without it?
